PetAdoptions/petsite/petsite/Controllers/AdoptionController.cs

```diff
@@ -86,0 +1,96 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+using PetSite.ViewModels;
+
+
+namespace PetSite.Controllers
+{
+    public class AdoptionController : BaseController
+    {
+        private readonly PetSite.Services.IPetSearchService _petSearchService;
+        private static Variety _variety = new Variety();
+        private readonly ILogger<AdoptionController> _logger;
+
+        public AdoptionController(ILogger<AdoptionController> logger, PetSite.Services.IPetSearchService petSearchService)
+        {
+            _petSearchService = petSearchService;
+            _logger = logger;
+        }
+        
+        // GET: Adoption
+        [HttpGet]
+        public IActionResult Index([FromQuery] Pet pet)
+        {
+            if (EnsureUserId()) return new EmptyResult(); // Redirect happened, stop processing
+            
+            // Check if pet data exists in TempData (from TakeMeHome redirect)
+            if (TempData["SelectedPet"] != null)
+            {
+                var petJson = TempData["SelectedPet"].ToString();
+                pet = JsonSerializer.Deserialize<Pet>(petJson);
+            }
+            
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Mass assignment detected. Attacker can exploit this auto-binding vulnerability to execute over-posting attacks. 
This attacks can lead to data corruption or unauthorised access. To prevent binding of unwanted properties, 
use `Bind` attribute on model to bind specific model properties.